source/postprocess/timing_statistics.cc

```diff
@@ -30,8 +30,15 @@ namespace aspect
     std::pair<std::string,std::string>
     TimingStatistics<dim>::execute (TableHandler &statistics)
     {
-      const auto &timer = this->get_computing_timer();
+      auto &timer = this->get_computing_timer();
+
+      // The timer only allows access to its data outside of subsections
+      // Leaving and entering the subsection again here means there will be
+      // one additional call of this subsection per time step in the timing
+      // output.
+      timer.leave_subsection("Postprocessing");
       const std::map<std::string, double> &timing_map = timer.get_summary_data(TimerOutput::total_wall_time);
+      timer.enter_subsection("Postprocessing");
 
       for (const auto &section: timing_map)
```

Comment:
I don't think it matters very much, but one would at least conceptually think that if you should only access the map outside any section, the right approach would be to take a *copy* rather than a *reference* here. In other words, you go out of the section, take a *snapshot* of the state, and then go back into the section to let it to its thing. This might be marginally slower (copying costs time), but I don't think it is observable and would seem like a slightly cleaner approach. If you think the same way, tell me and I'll make that change.

Reply:
See #6730.